### PR TITLE
Backend/seed database script

### PR DESCRIPTION
**Pull request by andrewyguo**
## Notion ticket link

[Seed Database](https://www.notion.so/uwblueprintexecs/Write-script-to-seed-database-b8f9b72b2f9e4f64ac45258e4e63d884)

## Implementation description
* Create shell script to seed the users, stories, and story_translations data bases  
* Also has option to clear the three data bases 

## Steps to test
1. Go into the root directory of `planet-read` and run `./seed_db.sh erase`
2. Set the variables `ADMIN_AUTH_ID` and `USER_AUTH_ID` to their respective values (can be found in the **planetread-secrets** channel) 
3. After data base has been cleared, run `./seed_db.sh` 

## What should reviewers focus on?
* That the script successfully generates 7 entries for `users`, 7 entries for `stories`, and 21 entries for `story_translations `

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: seed_db.sh [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Alexander', 'Hamilton', '$USER_AUTH_ID', 'User', '{\"ENGLISH\":3}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Angela', 'Merkel', '$ADMIN_AUTH_ID', 'Admin', '{\"GERMAN\":10}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Richard', 'Feynman', '$USER_AUTH_ID', 'User', '{\"PORTUGESE\":3}');"
+
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "ALTER SEQUENCE stories_id_seq RESTART WITH 1;"
+    
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO stories (title, description, youtube_link, level, translated_languages) VALUES ('Kevin Burns Coal to Create Smog', 'He wants to test out the HEPA filter in the new Tesla he got for his birthday', 'link', 4, '{\"GERMAN\", \"ENGLISH_UK\"}');"
```
Comment (gaoxk) [inline seed_db.sh:55]: Can you use real youtube links? That way the story cards load real youtube iframes. Feel free to use random youtube videos.

File: seed_db.sh [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Miroslav', 'Klose', '$USER_AUTH_ID', 'User', '{\"POLISH\":5, \"GERMAN\":3}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Kevin', 'De Bryune', '$USER_AUTH_ID', 'User', '{\"DUTCH\":5, \"FRENCH\":6}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Dwight', 'D. Eisenhower', '$USER_AUTH_ID', 'User', '{\"ENGLISH\":10}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Alexander', 'Hamilton', '$USER_AUTH_ID', 'User', '{\"ENGLISH\":3}');"
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('Angela', 'Merkel', '$ADMIN_AUTH_ID', 'Admin', '{\"GERMAN\":10}');"
```
Comment (gaoxk) [inline seed_db.sh:50]: Let's keep all possible levels between 1 and 4

File: seed_db.sh [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+#           not NGO-friendly. Please use this option cautiously. 
+
+
+
+if [ -z "$USER_AUTH_ID" ] 
+then 
+    echo "Warning: \$USER_AUTH_ID is empty. Please set the value to the one posted in the secrets channel." 
+fi 
+
+if [ -z "$ADMIN_AUTH_ID" ]
```
Comment (gaoxk) [inline seed_db.sh:24]: i think the script should exit if these AUTH_IDs are missing, unless the `erase` command is being passed in.

File: seed_db.sh [hunk 4]
```diff
@@ -0,0 +1,136 @@
+# ***** How to Use: *****
+# For now, you will need to set the two secret variables $USER_AUTH_ID and $ADMIN_AUTH_ID in your own shell.
+# The values of these variables can be found in the secret channel. 
+# Working to add these to vault and, in a future update, you won't need to set these variables manually. 
+# 
+# In the directory of the script, call ./seed_db.sh [OPTION]
```
Comment (gaoxk) [inline seed_db.sh:6]: nit: Just to make things crystal clear, can you add an example like

`USER_AUTH_ID=123 ADMIN_AUTH_ID=1234 ./seed_db.sh kevin`

**Review by gaoxk — CHANGES_REQUESTED** [review]
lookin good 😎  can you send a message to the dev slack channel about the CRLF vs LF issue we ran into? Seems like some folks will have to switch the file ending type to make it work.

**Comment by gaoxk** [comment]
Discussion from the discord conversation:
- make new firebase users with easy to remember passwords/emails (eg planetread+angelamerkel@uw..)
- once this pr is merged, send message to secrets channel of 
  - generated user email & passwords
  - all environment variables set on one line to make copy paste ezpz for developers
- make sure inserted `story_translations` line up with their respectiove `story.translated_languages`

**Review by gaoxk — CHANGES_REQUESTED** [review]
Looks good! lets merge after two small nits :)

**Comment by gaoxk** [comment]
one more thing: can you insert some `story_content `+ respective `story_translation_content `for each of the stories and story translations? feel free to make a loop making incoherent garbage text to keep the script concise.

File: seed_db.sh [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (6, 'ENGLISH_US', 'START', 3);"
+
+
+    # Story content 
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "ALTER SEQUENCE story_contents_id_seq RESTART WITH 1;"
+
+    for STORY_ID in 1 2 3 4 5 6 7
+    do 
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_contents (story_id, line_index, content) VALUES ('$STORY_ID', 1, '“Every two weeks I went to a meeting with them, and in my room here I covered pages with writing.
```
Comment (e-wai) [inline seed_db.sh:138]: Hey, I'm using some of the script manually atm: 

` ('$STORY_ID', 1, '“Every ` -> what's the `“` for?

File: seed_db.sh [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            That says the way is open. That throws it right back on a man.
+            For if ‘Thou mayest’—it is also true that ‘Thou mayest not.’ Don’t you see?”');"
+    done 
+
+    
+    # Story translation content 
+    docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "ALTER SEQUENCE story_translation_contents_id_seq RESTART WITH 1;"
+
+
+    for STORY_TRANSLATION_ID in 2 3 5 7 11 13
```
Comment (gaoxk) [inline seed_db.sh:175]: For every story_translation that exists, the respective story_translation_content lines must also exist (see [create_story_translation](https://github.com/uwblueprint/planet-read/blob/main/backend/python/app/services/implementations/story_service.py#L85) service). i.e. if you create a story translation of a 6 line story, the there must also be 6 lines of story translation content created with  empty strings. 

A quick fix is to run this forloop from 1 to 13. To diversify our data, you can make one story translation full of emtpy strings, and one that has some empty and some full lines.

**Review by gaoxk — CHANGES_REQUESTED** [review]

File: seed_db.sh [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+        
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 10, 'But the Hebrew word, the word timshel—‘Thou mayest’—that gives a choice. 
+            It might be the most important word in the world. 
+            That says the way is open. That throws it right back on a man.
+            For if ‘Thou mayest’—it is also true that ‘Thou mayest not.’ Don’t you see?”');"
+    done 
+
+    for STORY_TRANSLATION_ID in 1 4 6 8 9 10  
+    do 
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 1, '');"
```
Comment (gaoxk) [inline seed_db.sh:212]: nit: you can nest a `for i in 1 2 3 4 5 6 7 8 9 10` and use `i` as the `line_index` here to reduce the repeat code.

File: seed_db.sh [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 7, '');"
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 8, '');"
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 9, '');"
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 10, '');"
+    done 
+
+    for STORY_TRANSLATION_ID in 12 
+    do 
+        docker exec -it planet-read_db_1 psql -U postgres -d planet-read -c "INSERT INTO story_translation_contents (story_translation_id, line_index, translation_content) VALUES ('$STORY_TRANSLATION_ID', 1, '');"
```
Comment (gaoxk) [inline seed_db.sh:226]: ditto nit

**Review by gaoxk — CHANGES_REQUESTED** [review]

**Review by gaoxk — APPROVED** [review]

**Pull request merged**